src/lsp-server.ts

```diff
@@ -1123,4 +1155,71 @@ export class LspServer {
             ...workspacePreference.inlayHints ?? {}
         };
     }
+
+    async semanticTokensFull(params: lsp.SemanticTokensParams): Promise<lsp.SemanticTokens> {
+        const file = uriToPath(params.textDocument.uri);
+        this.logger.log('semanticTokensFull', params, file);
+        if (!file) {
+            return { data: [], resultId: '' };
+        }
+
+        const doc = this.documents.get(file);
+        if (!doc) {
+            return { data: [], resultId: '' };
+        }
+
+        const start = doc.offsetAt({
+            line: 0,
+            character: 0
+        });
+        const end = doc.offsetAt({
+            line: doc.lineCount,
+            character: 0
```

Comment:
Does this range goes beneath the actual `end` to simplify things? If so then I guess it's not entirely correct but if tsserver handles that correctly then probably we can keep it.

Could you create at least one simple test that verifies that range request covers whole document, including the the whole last line?

Reply:
Never mind, it looks like this will goes past the actual end of the document potentially but if tsserver handles that fine then I don't mind.

Reply:
Actually the spec even says:

>  If you want to specify a range that contains a line including the line ending character(s) then use an end position denoting the start of the next line.

so it's all good.